src/class/Model.ts

```
@@ -29,13 +29,78 @@ export class Model {
   static unique?: Array<string[]>;
   static primaryKey?: string[];
   private static sql = ''; // stores current db query here until done executing
+  private static openTran = false; // flag for whether transaction is on first run or not
+  private static tranFailed = false; // flag for whether query in a transaction has failed
+  private static connectTran?: PoolClient;
   static foreignKey?: {
     columns: string[];
     mappedColumns: string[];
     rel_type?: string;
     table: string;
   }[];
 
+  /**
+   * params: uri | void
+   * Begins and continues adding queries to the transaction.
+   *
+   * Note: Initially checking the tranFailed state solves the issue of duplicate error handling messages appearing
+   */
+  static async tran(uri?: string) {
+    if (!Model.tranFailed) {
+      const db = Model.connectTran;
+      if (!Model.openTran) {
+        this.sql = 'BEGIN;' + this.sql;
+        // create connection to db
+        Model.connectTran = await ConnectDb(uri);
+        try {
+          await db.queryObject(this.sql);
+          Model.openTran = true;
+        } catch (err) {
+          Model.tranFailed = true;
+        }
+      } else {
+        try {
+          await db.queryObject(this.sql);
+        } catch (err) {
+          Model.tranFailed = true;
+          console.log(err, 'ERROR IN TRANSACTION');
```

Do we want to throw an error or exception here instead of a console log? If we throw an exception then the user can make logical decisions based on their catch since this is asynchronous.